core/src/main/scala/com/mesosphere/usi/core/Scheduler.scala

```diff
@@ -43,44 +50,84 @@ import scala.concurrent.Future
  *                                     +----------------------+
  */
 object Scheduler {
-  case class MesosConnection(mesosHostName: String)
+  type SpecInput = (SpecsSnapshot, Source[SpecUpdated, Any])
 
-  def connect(mesosHostName: String): Future[(MesosConnection, Flow[SpecEvent, StateEvent, NotUsed])] = {
-    val flow = Flow.fromGraph {
-      GraphDSL.create(Scheduler.unconnectedGraph, FakeMesos.flow)((_, _) => NotUsed) { implicit builder =>
-        { (graph, mockMesos) =>
+  type StateOutput = (StateSnapshot, Source[StateEvent, Any])
+
+  case class MesosConnection(mesosHostName: String, mesosCallFactory: MesosCalls, frameworkId: FrameworkID, schedulerFlow: Flow[SpecInput, StateOutput, NotUsed])
+
+  // TODO - provide abstraction for FrameworkID persistence, recovery, etc.; we'll want to exclude the frameworkId
+  // TODO - This layer should provide a cancellation mechanism and an auto-reconnect mechanism
+  def connect(settings: MesosClientSettings, frameworkInfo: FrameworkInfo)(
+      implicit actorSystem: ActorSystem,
+      materializer: ActorMaterializer): Future[MesosConnection] = {
+
+    import scala.concurrent.ExecutionContext.Implicits.global
+
+    MesosClient(settings, frameworkInfo).runWith(Sink.head).map { client =>
+      val flow = mesosConnectedGraph(client.calls, Flow.fromSinkAndSource(client.mesosSink, client.mesosSource))
+      MesosConnection(settings.master, client.calls, client.frameworkId, flow)
+    }
+  }
+
+  def mesosConnectedGraph(mesosCallFactory: MesosCalls, mesosFlow: Flow[MesosCall, MesosEvent, Any]): Flow[SpecInput, StateOutput, NotUsed] = {
+    Flow.fromGraph {
+      GraphDSL.create(unconnectedGraph(mesosCallFactory), mesosFlow)((_, _) => NotUsed) { implicit builder =>
+        { (graph, mesos) =>
           import GraphDSL.Implicits._
 
-          mockMesos ~> graph.in2
-          graph.out2 ~> mockMesos
+          mesos ~> graph.in2
+          graph.out2 ~> mesos
 
           FlowShape(graph.in1, graph.out1)
         }
       }
     }
-    Future.successful(MesosConnection(mesosHostName) -> flow)
   }
 
-  /*
-   */
-  def unconnectedGraph: BidiFlow[SpecEvent, StateEvent, Mesos.Event, Mesos.Call, NotUsed] = {
+  /**
+    * We expess an interface of receiving a snapshot, followed by a series of events, to guide consumers of USI down a
+    * proper implementation path. A SpecSnapshot should be the very first thing that the USI Scheduler receives from the
+    * Framework implementation.
+    *
+    * However, for convenience in working with streams, internally we deal with a single stream of SpecEvents.
```

Comment:
I imagined that scheduler would start Mesos task reconciliation after receiving the `SpecSnapshot` since this is a good point in time when all the reconciled tasks should be ideally found in the snapshot. Wdty?

Reply:
I believe it should be the case as well. But I think reconciliation is out of scope for now

Reply:
Yes, https://jira.mesosphere.com/browse/DCOS-47476

Reply:
`//TODO:  DCOS-48503`